app/models/time_profile.rb

```diff
@@ -33,6 +33,19 @@ def self.seed
     end
   end
 
+  def self.with_cached_time_profile_ids(profile = nil)
```

Comment:
My only concern here is that we don't pass in a timezone, in fact we pass a nil tz or timeprofile to default_time_profile.  Is it possible that we are caching time profile ids that aren't relevant for a timezone?  🤔